acme.go

```diff
@@ -140,13 +148,13 @@ func Register(client *http.Client, config *Config) error {
 // which the client will have to choose from and perform, to complete authorization.
 //
 // If client argument is nil, DefaultClient will be used.
-func authorize(client *http.Client, config *Config, domain string) (ChallengeSet, error) {
+func authorize(client *http.Client, config *Config, domain string) (Authorization, error) {
```

Comment:
to me the rule here is something like if it's a struct, always return a pointer unless it's small and you 99% sure it'll never change.